packages/twenty-server/src/engine/api/graphql/graphql-query-runner/resolvers/graphql-query-merge-many-resolver.service.ts

```diff
@@ -218,6 +243,238 @@ export class GraphqlQueryMergeManyResolverService extends GraphqlQueryBaseResolv
+          priorityRecord,
+          propertyName,
+          recordsWithValues,
+        );
+
+        return;
+      }
+
+      if (property.isArray || property.type === FieldMetadataType.RAW_JSON) {
+        mergedComposite[propertyName] = this.mergeArrayProperty(
+          recordsWithValues,
+          propertyName,
+          compositeDefinition,
+        );
+
+        return;
+      }
+
+      mergedComposite[propertyName] = this.mergeSimplePropertyFromAllRecords(
+        priorityRecord,
+        recordsWithValues,
+        propertyName,
+      );
+    });
+
+    return mergedComposite;
+  }
+
+  private getUniqueConstraintValue(
+    priorityRecord: { value: unknown; recordId: string } | undefined,
+    propertyName: string,
+    recordsWithValues?: { value: unknown; recordId: string }[],
+  ): unknown {
+    const priorityValue = priorityRecord?.value as Record<
+      string,
+      unknown
+    > | null;
+
+    const priorityFieldValue = priorityValue?.[propertyName];
+
+    if (hasRecordFieldValue(priorityFieldValue)) {
+      return priorityFieldValue;
+    }
+
+    if (recordsWithValues) {
+      for (const record of recordsWithValues) {
+        const recordValue = record.value as Record<string, unknown> | null;
+        const fieldValue = recordValue?.[propertyName];
+
+        if (hasRecordFieldValue(fieldValue)) {
+          return fieldValue;
+        }
+      }
+    }
+
+    return null;
+  }
+
+  private mergeArrayProperty(
+    recordsWithValues: { value: unknown; recordId: string }[],
+    propertyName: string,
+    compositeDefinition: CompositeType,
+  ): unknown[] | null {
+    const allArrays: unknown[] = [];
+
+    recordsWithValues.forEach((record) => {
+      const recordValue = record.value as Record<string, unknown> | null;
+      const arrayValue = recordValue?.[propertyName];
+
+      if (Array.isArray(arrayValue)) {
+        allArrays.push(...arrayValue);
+      }
+    });
+
+    const uniqueArrays = this.deduplicateCompositeArrayItems(
+      allArrays,
+      compositeDefinition,
+    );
+
+    return uniqueArrays.length > 0 ? uniqueArrays : null;
+  }
+
+  private mergeSimplePropertyFromAllRecords(
+    priorityRecord: { value: unknown; recordId: string } | undefined,
+    recordsWithValues: { value: unknown; recordId: string }[],
+    propertyName: string,
+  ): unknown {
+    const priorityValue = priorityRecord?.value as Record<
+      string,
+      unknown
+    > | null;
+
+    const priorityFieldValue = priorityValue?.[propertyName];
+
+    if (hasRecordFieldValue(priorityFieldValue)) {
+      return priorityFieldValue;
+    }
+
+    for (const record of recordsWithValues) {
+      const recordValue = record.value as Record<string, unknown> | null;
+      const fieldValue = recordValue?.[propertyName];
+
+      if (hasRecordFieldValue(fieldValue)) {
+        return fieldValue;
+      }
+    }
+
+    return null;
+  }
+
+  private isArrayLikeField(fieldType: FieldMetadataType): boolean {
```

Comment:
let's avoid this and write a big switch on field metadata type (no need for abstraction)